securedrop_export/export.py

```diff
@@ -277,31 +278,42 @@ def unlock_luks_volume(self, encryption_key):
                 if 'UUID' in items[0]:
                     self.encrypted_device = 'luks-' + items[1]
 
-            # the luks device is not already unlocked
-            if not os.path.exists(os.path.join("/dev/mapper/", self.encrypted_device)):
-                logger.debug('Unlocking luks volume {}'.format(self.encrypted_device))
-                p = subprocess.Popen(
-                    ["sudo", "cryptsetup", "luksOpen", self.device, self.encrypted_device],
-                    stdin=subprocess.PIPE,
-                    stdout=subprocess.PIPE,
-                    stderr=subprocess.PIPE
-                )
-                logger.debug('Passing key')
-                p.communicate(input=str.encode(encryption_key, "utf-8"))
-                rc = p.returncode
-                if rc != 0:
-                    logger.error('Bad phassphrase for {}'.format(self.encrypted_device))
-                    self.exit_gracefully(ExportStatus.USB_BAD_PASSPHRASE.value)
+            # the luks device is already unlocked
+            if os.path.exists(os.path.join('/dev/mapper/', self.encrypted_device)):
+                logger.debug('Device already unlocked')
+                return
+
+            logger.debug('Unlocking luks volume {}'.format(self.encrypted_device))
+            p = subprocess.Popen(
+                ["sudo", "cryptsetup", "luksOpen", self.device, self.encrypted_device],
+                stdin=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE
+            )
+            logger.debug('Passing key')
+            p.communicate(input=str.encode(encryption_key, "utf-8"))
+            rc = p.returncode
+            if rc != 0:
+                logger.error('Bad phassphrase for {}'.format(self.encrypted_device))
+                self.exit_gracefully(ExportStatus.USB_BAD_PASSPHRASE.value)
         except subprocess.CalledProcessError:
             self.exit_gracefully(ExportStatus.USB_ENCRYPTION_NOT_SUPPORTED)
 
     def mount_volume(self):
+        # If the drive is already mounted then we don't need to mount it again
+        output = subprocess.check_output(
+            ["lsblk", "-o", "MOUNTPOINT", "--noheadings", self.device])
+        mountpoint = output.decode('utf-8').rstrip()
+        if mountpoint:
+            logger.debug('The device is already mounted')
+            self.mountpoint = mountpoint
+            return
+
         # mount target not created, create folder
-        if not os.path.exists(self.mountpoint):
-            self.safe_check_call(
-                command=["sudo", "mkdir", self.mountpoint],
-                error_message=ExportStatus.ERROR_USB_MOUNT
-            )
+        self.safe_check_call(
+            command=["sudo", "mkdir", self.mountpoint],
```

Comment:
since `self.mountpoint` is hardcoded to be `/media/usb` unless the drive is already mounted, what happens if a directory called `/media/usb` already exists? (I just hit this case while testing this)

Reply:
ah good find. fixing